examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart

```diff
@@ -0,0 +1,158 @@
+                                ),
+                              )
+                              .toList(),
+                        ),
+                      ),
+                    ],
+                  ),
+                  icon: Icons.pets,
+                ),
+                PuppyEditCard(
+                  label: 'Gender',
+                  content: Column(
+                    children: [
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                        children: [
+                          Row(
+                            children: [
+                              const Text(
+                                'Male',
+                                style: TextStyles.editableTextStyle,
+                              ),
+                              Radio<Gender>(
+                                key: const ValueKey('PuppyGenderMaleRadio'),
+                                value: Gender.Male,
+                                groupValue: _viewModel.puppy.gender,
+                                onChanged: (gender) =>
+                                    _viewModel.onGenderChange(gender!),
+                              ),
+                            ],
+                          ),
+                          Row(
+                            children: [
+                              const Text(
+                                'Female',
+                                style: TextStyles.editableTextStyle,
+                              ),
+                              Radio<Gender>(
+                                key: const ValueKey('PuppyGenderFemaleRadio'),
+                                value: Gender.Female,
+                                groupValue: _viewModel.puppy.gender,
+                                onChanged: (gender) =>
+                                    _viewModel.onGenderChange(gender!),
+                              ),
+                            ],
+                          ),
+                        ],
+                      ),
+                    ],
+                  ),
+                  icon: Icons.wc,
+                ),
+                PuppyEditCard(
+                  label: 'Characteristics',
+                  content: TextFormField(
+                    key: const ValueKey('PuppyCharacteristicsInputField'),
+                    maxLines: 8,
+                    initialValue: _viewModel.puppy.breedCharacteristics,
+                    cursorColor: const Color(0xff333333),
+                    style: TextStyles.editableTextStyle,
+                    textInputAction: TextInputAction.done,
+                    maxLengthEnforcement: MaxLengthEnforcement.enforced,
+                    decoration: InputStyles.textFieldDecoration,
+                    onChanged: (characteristics) =>
+                        _viewModel.onCharacteristicsChange(characteristics),
+                    validator: (value) => (value == null || value.isEmpty)
```

Comment:
@GoranPrime, please move all validation logic to the business layer

Reply:
@StanevPrime Corrected